tests/test_model_builder.py

```diff
@@ -0,0 +1,294 @@
+TNRedAmp -14.227505410948254
+TNRedGam 4.91353
+TNRedC 45
+T2EFAC -f L-wide_PUPPI   1.507
+T2EFAC -f 430_ASP   1.147
+T2EQUAD -f L-wide_PUPPI   0.25518
+T2EQUAD -f 430_ASP   0.01410
+ECORR -f 430_PUPPI   0.00601
+ECORR -f L-wide_PUPPI   0.31843
+JUMP -fe L-wide      -0.000009449  1       0.000009439
+"""
+
+
+class SimpleModel(PhaseComponent):
+    """Very simple test model component
+    """
+
+    register = True
+    category = "simple_test"
+
+    def __init__(self):
+        super(SimpleModel, self).__init__()
+        self.add_param(floatParameter(name="TESTPARAM", value=0.0, unit="s"))
+
+
+class SubsetModel(PhaseComponent):
+    """Test model component hosting the parameters which are a subset of spindown.
+    """
+
+    register = False  # This has to be false, otherwrise all test will fail.
+    category = "simple_test"
+
+    def __init__(self):
+        super(SubsetModel, self).__init__()
+        self.add_param(floatParameter(name="F0", value=0.0, unit="1/s"))
+        self.add_param(floatParameter(name="F1", value=0.0, unit="1/s^2"))
+
+
+@pytest.fixture
+def sub_set_model():
+    return SubsetModel()
+
+
+@pytest.fixture
+def simple_model():
+    return SimpleModel()
+
+
+@pytest.fixture
+def simple_model_overlap(simple_model):
+    simple_model.add_param(floatParameter(name="F0", aliases=[], value=0.0, unit="1/s"))
+    return simple_model
+
+
+@pytest.fixture
+def simple_model_alias_overlap():
+    simple_model = SimpleModel()
+    simple_model.add_param(
+        floatParameter(name="TESTPARAM2", aliases=["F0"], value=0.0, unit="1/s")
+    )
+    simple_model.add_param(
+        floatParameter(name="TESTPARAM3", aliases=["LAMBDA"], value=0.0, unit="deg")
+    )
+    return simple_model
+
+
+def test_model_builder_class():
+    """Test if model builder collected components information correctly
+    """
+    mb = ModelBuilder()
+    category = mb.category_component_map
+    assert len(mb.param_component_map["PX"]) == len(category["astrometry"])
+    assert len(mb.component_category_map) == len(mb.components)
+    assert len(mb.param_alias_map) == len(mb.param_component_map)
+    # test for new components
+    assert "SimpleModel" in mb.components
+    simple_comp = mb.components["SimpleModel"]
+    simple_comp.add_param(
+        floatParameter(name="TESTPARAM2", aliases=["F0"], value=0.0, unit="s")
+    )
+
+
+def test_aliases_mapping():
+    """Test if aliases gets mapped correclty
+    """
+    mb = ModelBuilder()
+    # all alases should be mapped to the components
+    assert len(mb.param_alias_map) == len(mb.param_component_map)
+
+    # Test if the param_alias_map is passed by pointer
+    _ = mb._add_alias_to_map("TESTAX", "TESTAXX", mb.param_alias_map)
+    assert "TESTAX" in mb.param_alias_map
+    # Test existing entry
+    _ = mb._add_alias_to_map("F0", "F0", mb.param_alias_map)
+    assert mb.param_alias_map["F0"] == "F0"
+    # Test repeatable_params with differnt indices.
+    for rp in mb.repeatable_param:
```

Comment:
Can you make the repeatable params test a separate test? I'm not quite sure I understand what repeatable parameters are.